### PR TITLE
Added JumpType indicator to AssemblyItem.

### PR DESCRIPTION
**Pull request by LianaHus**

### CODE REVIEW

File: libevmcore/Assembly.h [hunk 1]
```diff
@@ -58,13 +60,17 @@ class AssemblyItem
 	int deposit() const;
 
 	bool match(AssemblyItem const& _i) const { return _i.m_type == UndefinedItem || (m_type == _i.m_type && (m_type != Operation || m_data == _i.m_data)); }
-	void setLocation(SourceLocation const& _location) { m_location = _location;}
+	void setLocation(SourceLocation const& _location) { m_location = _location; }
 	SourceLocation const& getLocation() const { return m_location; }
 
+	void setJumpAnnotationType(JumpAnnotationType const& _jumpAnnotation) { m_jumpAnnotation = _jumpAnnotation; }
```
Comment (chriseth) [inline libevmcore/Assembly.h]: Enum can (and I think should) be passed by value.

File: libevmcore/Assembly.h [hunk 2]
```diff
@@ -58,13 +60,17 @@ class AssemblyItem
...
+	JumpAnnotationType const& getJumpAnnotationType() const { return m_jumpAnnotation; }
+
 private:
 	AssemblyItemType m_type;
 	u256 m_data;
 	SourceLocation m_location;
+	JumpAnnotationType m_jumpAnnotation;
```
Comment (chriseth) [inline libevmcore/Assembly.h]: just "jumpType" for both?

File: libevmcore/Assembly.cpp [hunk 3]
```diff
@@ -75,6 +75,20 @@ int AssemblyItem::deposit() const
 	return 0;
 }
 
+string AssemblyItem::getJumpTypeAsString() const
```
Comment (chriseth) [inline libevmcore/Assembly.cpp]: I think this should be just a single line inside `Assembly::streamRLP`.

Comment (LianaHus) [inline libevmcore/Assembly.cpp]: I thought this way it's clearer but I can change it if you want.

File: libsolidity/Compiler.cpp [hunk 4]
```diff
@@ -177,7 +177,9 @@ void Compiler::appendFunctionSelector(ContractDefinition const& _contract)
 	{
 		callDataUnpackerEntryPoints.insert(std::make_pair(it.first, m_context.newTag()));
 		m_context << eth::dupInstruction(1) << u256(FixedHash<4>::Arith(it.first)) << eth::Instruction::EQ;
-		m_context.appendConditionalJumpTo(callDataUnpackerEntryPoints.at(it.first));
+		auto assemblyItem = callDataUnpackerEntryPoints.at(it.first);
+		//assemblyItem.setJumpType(eth::AssemblyItem::JumpType::IntoFunction);
```
Comment (chriseth) [inline libsolidity/Compiler.cpp]: Comment

File: libsolidity/Compiler.cpp [hunk 5]
```diff
@@ -197,7 +199,9 @@ void Compiler::appendFunctionSelector(ContractDefinition const& _contract)
 		m_context << callDataUnpackerEntryPoints.at(it.first);
 		eth::AssemblyItem returnTag = m_context.pushNewTag();
 		appendCalldataUnpacker(functionType->getParameterTypes());
-		m_context.appendJumpTo(m_context.getFunctionEntryLabel(functionType->getDeclaration()));
+		auto assemblyItem = m_context.getFunctionEntryLabel(functionType->getDeclaration());
+		//assemblyItem.setJumpType(eth::AssemblyItem::JumpType::IntoFunction);
```
Comment (chriseth) [inline libsolidity/Compiler.cpp]: This, on the other hand, should not be a jump into a function, I would say (it is also likely to be optimized away).

File: libsolidity/Compiler.cpp (hunk 4 above)
Comment (LianaHus) [inline libsolidity/Compiler.cpp]: sorry the code is in progress, I'll clean it up

**Pull request merged**